lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -488,3 +489,37 @@ def plot_binding_histogram(self, kind, bins=10, **kwargs):
         plt.bar(edges[:-1], counts, width=widths, align="edge", **kwargs)
         plt.ylabel("Counts")
         plt.xlabel(f"Position ({self[0]._image._position_unit[1]})")
+
+    def _histogram_binding_profile(self, n_time_bins, bandwidth, n_position_points):
+        """Calculate a Kernel Density Estimate (KDE) of binding density along the tether for time bins.
+
+        First the kymograph is binned along the temporal axis. In the case of non-integer `frames / bins`,
+        remaining frames are discarded. Next, for each time bin, a KDE is calculated along the spatial axis.
+
+        Parameters
+        ----------
+        n_time_bins : int
+            Requested number of time bins.
+        bandwidth : float
+            KDE bandwidth; units are in the physical spatial units of the kymograph.
+        n_position_points : int
+            Length of the returned density array(s).
+        """
+
+        n_rows, n_frames = self[0]._image.data.shape
+        position_max = n_rows * self[0]._image._pixel_size
+        bin_size = n_frames // n_time_bins
```

Comment:
This fails when `n_time_bins > n_frames`, produces a division by zero when computing the bin edges in the next step.

Reply:
good catch, re-throw with a better error message and added a test